pkg/beacon/relay/registry/storage.go

```diff
@@ -25,15 +31,32 @@ func newStorage(persistence persistence.Handle) storage {
 	}
 }
 
-func (ps *persistentStorage) save(membership *Membership) error {
+func (ps *persistentStorage) save(
+	membership *Membership,
+	groupState GroupState,
+) error {
 	membershipBytes, err := membership.Marshal()
 	if err != nil {
 		return fmt.Errorf("marshalling of the membership failed: [%v]", err)
 	}
 
-	hexGroupPublicKey := hex.EncodeToString(membership.Signer.GroupPublicKeyBytesCompressed())
+	directory := hex.EncodeToString(membership.Signer.GroupPublicKeyBytesCompressed())
+
+	var membershipPrefix string
+	switch groupState {
```

Comment:
Not sure what will be the final solution around prefixes, but I think one way of not relying on the `switch` might be something like this:
```
// GroupState determines the given group state.
type GroupState string

// Enum that denotes possible group states.
const (
	GroupStateCandidate GroupState = "candidate"
	GroupStateApproved = "approved"
)
```

and then in the `storage.go` we can replace the `switch` with:

```
membershipPrefix := fmt.Sprintf("%v_%v", groupState, "membership")
```

Just an idea, unless we need integers for the state as our "enums".